drivers/issi/is31fl3733.c

```diff
@@ -80,43 +80,60 @@ bool    g_pwm_buffer_update_required[DRIVER_COUNT] = {false};
 uint8_t g_led_control_registers[DRIVER_COUNT][24]             = {{0}, {0}};
 bool    g_led_control_registers_update_required[DRIVER_COUNT] = {false};
 
-void IS31FL3733_write_register(uint8_t addr, uint8_t reg, uint8_t data) {
+uint8_t IS31FL3733_write_register(uint8_t addr, uint8_t reg, uint8_t data) {
+    // If the transaction fails function returns 1.
+    uint8_t output = 0;
     g_twi_transfer_buffer[0] = reg;
     g_twi_transfer_buffer[1] = data;
 
 #if ISSI_PERSISTENCE > 0
     for (uint8_t i = 0; i < ISSI_PERSISTENCE; i++) {
-        if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT) == 0) break;
+        if (i2c_transmit(addr << 1, g_twi_transfer_buffer, 2, ISSI_TIMEOUT) == 0) {
+            break;
+        } else {
+            output = 1;
```

Comment:
I think this might be cleaner just to return 1

Reply:
this is not going to work because of the for loop i am pretty sure. Thats why I went for this hacky way. Another way is to do a break and then return 1. But then if lets say transmit number 3 fails you will not transmit the rest either.

Reply:
when you get a 0, it will break out of the loop. You can return from a loop.